whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NotificationFileGenerationService.java

```diff
@@ -0,0 +1,43 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.persist.NrtmDocumentType;
+import net.ripe.db.nrtm4.persist.NrtmSource;
+import net.ripe.db.nrtm4.persist.NrtmVersionInformationDao;
+import net.ripe.db.nrtm4.persist.VersionInformation;
+import net.ripe.db.nrtm4.publish.SnapshotFile;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+
+@Service
+public class NotificationFileGenerationService {
+
+    private final NrtmVersionInformationDao versionDao;
+
+    public NotificationFileGenerationService(
+            final NrtmVersionInformationDao nrtmVersionInformationDao
+    ) {
+        versionDao = nrtmVersionInformationDao;
+    }
+
+    // TODO: Add a global lock to ensure that no other instance can run until this method exits
+    public SnapshotFile generateSnapshot(final NrtmSource source) {
```

Comment:
I know that we are using IT for testing it, but I think that for this kind of code (code with some logic) and add unit tests as well.

Reply:
i agree. i've done some refactoring in 'transform-rpsl-to-nrtm-json' to make it easier to unit test parts of the implementation. there could, and *should* be more though. unit test will be better than integration tests for some things, but they're also fragile coz they depend heavily on the code, and i'm not sure how all of that will look yet.

Reply:
yes, this is why I think that we should add (never replace) to IT in this case I mean